base_layer/core/src/consensus/consensus_constants.rs

```diff
@@ -121,19 +122,39 @@ impl ConsensusConstants {
 
     /// The amount of PoW algorithms used by the Tari chain.
     pub fn get_pow_algo_count(&self) -> u64 {
-        self.pow_algo_count
+        let mut count = 0;
```

Comment:
Case in point, if it were just the target block times, or straight splits, this would be one line:
`self.algo_split.len()`

Reply:
I am replaying to this below, as the single replay is covers both

Reply:
But more specifically, I dont think an array of target block times will just be able to say .len() As you might have 3 algorithms defined in your pow struct. Or you are running local_net with purely a single pow. Then you need to make it an vec, and make sure it only matches your single algo, meaning len = 0. 

But that does not solve the case we you are running blake , which is algo 2, how do you know which split to use.

If I understand your think correct here, if you only want to run blake. You would have an `algo_split = [120]`
But the get_target time function needs to know which index to look at, `algo_split[1] != [120]`, its going to do index out of bounds. So you have to do `algo_split = [0,120]`
But then `algo_split.len() != algo_count `